### ISSUE TITLE
Idea for script mode

### ISSUE THREAD

**Issue by jeroenvandijk** [dundalek/closh#96]
In #2 I had mentioned an option to use `eval` to support scripting mode. With cljs this is hard, now with jvm support this works. 

So here we go, after evaluating this in `closh`:

```clojure

(defn read-forms [blob]
  (->>
   ;; FIXME handle multine commands, how
   (clojure.string/split-lines blob)

   (remove (fn [form]
             ;; TODO decide on comment forms
             (or 
              (.startsWith form "#")
              (.startsWith form ";"))))))

(defn get-tokens [cmd-str]
  ;; FIXME i think there is a better than way to read a part of th ecommand
  (clojure.string/split cmd-str #"\s+"))

(defn exec-cmd [cmd-str]
  ;; Copied from sh in closh.zero.macros
  (eval (let [tokens (get-tokens cmd-str)]
          (-> (closh.zero.compiler/compile-interactive (closh.zero.parser/parse tokens))
              (closh.zero.pipeline/wait-for-pipeline)))))

(defn exec-script [script-location]
  (let [f (clojure.java.io/file script-location)]
    (when-not (.canExecute f)
      (throw (ex-info (str script-location " is not an executable") {})))
    
    (let [first-line (.readLine (java.io.BufferedReader. (java.io.FileReader. script-location)))]
      (if (.startsWith first-line "#!/usr/bin/env closh")
        ;; Huray for a new soul!
        (let [contents (slurp script-location)
              cmds (read-forms contents)]
          (doseq [cmd cmds]
            (exec-cmd cmd)))
        (shx script-location)))))

(spit "test.closh" (str "#!/usr/bin/env closh

ls | (pr-str) | sort | cat > foo.txt

ls |> (filter #(.startsWith % "C")) > bar.txt

")

chmod +x test.closh

(exec-script "test.closh")
```

I get errors like:
```
: command not found
: command not found
```
I don't have a clue yet how to evaluate stuff like this so I must be doing something weird. To be honest I don't know yet how to exactly use closh.. The script however does have an effect . So I consider this experiment to be semi-successful :)

**Comment by jeroenvandijk** [comment]
To do the evaluation above properly and safe (assuming the evaluation can work properly), maybe something like a `pod` as in `boot` is necessary https://github.com/boot-clj/boot/wiki/Pods

**Comment by dundalek** [comment]
As mentioned in #2 I haven't spent any time to make this work so any scripting that works is purely accidental :) But I am starting to see this would likely benefit to people in a relative short time frame, so I think I will focus more more in this direction.

My rough idea is that the script mode would parse the input and pass it down to the compiler in one piece so that any compiler errors would throw before executing any of the form. Basically like lumo or clj.main do things. Also the shebang execution needs to work.

**Comment by dundalek** [comment]
Initial iteration of script mode has now landed in the latest release v0.4, you can read how to use it in the [scripting docs](https://github.com/dundalek/closh/blob/master/doc/scripting.md). I will be glad for any feedback, experience or issue reports.
